assets/scripts/streets/StreetNameCanvas.jsx

```diff
@@ -85,15 +84,9 @@ class StreetNameCanvas extends React.Component {
     return classNames
   }
 
-  onClickStreetName = () => {
-    if (!this.props.editable) return
-
-    const newName = window.prompt(t('prompt.new-street', 'New street name:'), this.props.street.name)
-
-    if (newName) {
-      const name = StreetName.normalizeStreetName(newName)
-      this.props.saveStreetName(name, true)
-    }
+  handleStreetNameChange = (newName) => {
```

Comment:
Since most editing logic is being done in the `<StreetName>` component, let's consider moving the handler function into that component, which simplifies this component.